cpp/oneapi/dal/algo/decision_forest/common.cpp

```diff
@@ -16,6 +16,11 @@
 
 #include "oneapi/dal/algo/decision_forest/common.hpp"
 #include "oneapi/dal/algo/decision_forest/detail/model_impl.hpp"
+#include "oneapi/dal/exceptions.hpp"
+
+#define DAL_CHECK_DOMAIN_COND(cond, description) \
+    if (!(cond))                                 \
+        throw domain_error(description);
```

Comment:
Why you can't use function here?

Reply:
We would need to have refactoring iteration and introduce internal API for validation like @makart19 used here

Reply:
replaced with function